nibabel/tests/test_wrapstruct.py

```diff
@@ -485,3 +486,12 @@ class MyHdr(self.header_class):
         hdr['an_integer'] = 10
         s1 = str(hdr)
         assert_true('<unknown code 10>' in s1)
+        # Unencodable bytes are handled differently in Python 2 and 3
+        hdr['a_str'] = b'\xff'
+        s3 = str(hdr)
+        assert_true("a_str       : b'\\xff'" in s3 or
+                    "a_str       : \xff" in s3)
```

Comment:
Since we know what to expect for each version, seems better to have an `if / else` statement, based on the `sys.version_info`.